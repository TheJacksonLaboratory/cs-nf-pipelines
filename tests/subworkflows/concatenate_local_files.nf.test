nextflow_workflow {

    name "Test Workflow CONCATENATE_LOCAL_FILES"
    script "subworkflows/concatenate_local_files.nf"
    workflow "CONCATENATE_LOCAL_FILES"

    test("Paired-End Data") {

        when {
            params {
                pipeline = 'wgs'
                pubdir = "tests/results"
                read_type = "PE"
            }
            workflow {
                """
                // define inputs of the workflow here. Example:
                input[0] = Channel.fromList([
                    tuple(
                        'testSample_42',
                        [sampleID:'testSample_42', lane:'lane1', replicate:'NA', id:'testSample_42', size:1], 
                        'https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/mm10_WGS_sample_R1.fastq.gz', 
                        'https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/mm10_WGS_sample_R2.fastq.gz'
                    ),
                    tuple(
                        'testSample_101',
                        [sampleID:'testSample_101', lane:'lane1', replicate:'NA', id:'testSample_101', size:1], 
                        'https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/human/hg38_WGS_sample_R1.fastq.gz', 
                        'https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/human/hg38_WGS_sample_R2.fastq.gz'
                    )
                ])
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

    test("Single-End Data") {

        when {
            params {
                pipeline = 'wgs'
                pubdir = "tests/results"
                read_type = "SE"
            }
            workflow {
                """
                // define inputs of the workflow here. Example:
                input[0] = Channel.fromList([
                    tuple(
                        'testSample_42',
                        [sampleID:'testSample_42', lane:'lane1', replicate:'NA', id:'testSample_42', size:1], 
                        'https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/mm10_WGS_sample_R1.fastq.gz'
                    ),
                    tuple(
                        'testSample_101',
                        [sampleID:'testSample_101', lane:'lane1', replicate:'NA', id:'testSample_101', size:1], 
                        'https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/mm10_WGS_sample_R2.fastq.gz'
                    ),
                ])
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

    test("Multiple Lanes Paired-End Data") {

        when {
            params {
                pipeline = 'wgs'
                pubdir = "tests/results"
                read_type = "PE"
            }
            workflow {
                """
                // define inputs of the workflow here. Example:
                input[0] = Channel.fromList([
                    tuple(
                        'testSample_42',
                        [sampleID:'testSample_42', lane:'lane1', replicate:'NA', id:'testSample_42', size:2], 
                        'https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/individual_lanes/mm10-WGS-sample_GT42-42424_TTTTTTTT-TTTTTTTT_S42_L001_R1.fastq.gz',
                        'https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/individual_lanes/mm10-WGS-sample_GT42-42424_TTTTTTTT-TTTTTTTT_S42_L001_R2.fastq.gz'
                    ),
                    tuple(
                        'testSample_42',
                        [sampleID:'testSample_42', lane:'lane2', replicate:'NA', id:'testSample_42', size:2], 
                        'https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/individual_lanes/mm10-WGS-sample_GT42-42424_TTTTTTTT-TTTTTTTT_S42_L002_R1.fastq.gz',
                        'https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/individual_lanes/mm10-WGS-sample_GT42-42424_TTTTTTTT-TTTTTTTT_S42_L002_R2.fastq.gz'
                    )
                ])
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

    test("Multiple Lanes Single-End Data") {

        when {
            params {
                pipeline = 'wgs'
                pubdir = "tests/results"
                read_type = "SE"
            }
            workflow {
                """
                // define inputs of the workflow here. Example:
                input[0] = Channel.fromList([
                    tuple(
                        'testSample_42',
                        [sampleID:'testSample_42', lane:'lane1', replicate:'NA', id:'testSample_42', size:2], 
                        'https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/individual_lanes/mm10-WGS-sample_GT42-42424_TTTTTTTT-TTTTTTTT_S42_L001_R1.fastq.gz'
                    ),
                    tuple(
                        'testSample_42',
                        [sampleID:'testSample_42', lane:'lane2', replicate:'NA', id:'testSample_42', size:2], 
                        'https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/individual_lanes/mm10-WGS-sample_GT42-42424_TTTTTTTT-TTTTTTTT_S42_L001_R2.fastq.gz'
                    )
                ])
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
