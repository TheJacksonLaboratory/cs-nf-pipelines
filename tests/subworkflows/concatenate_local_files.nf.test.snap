{
    "Single-End Data": {
        "content": [
            {
                "0": [
                    [
                        "testSample_101",
                        {
                            "sampleID": "testSample_101"
                        },
                        [
                            "https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/mm10_WGS_sample_R2.fastq.gz"
                        ]
                    ],
                    [
                        "testSample_42",
                        {
                            "sampleID": "testSample_42"
                        },
                        [
                            "https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/mm10_WGS_sample_R1.fastq.gz"
                        ]
                    ]
                ],
                "read_meta_ch": [
                    [
                        "testSample_101",
                        {
                            "sampleID": "testSample_101"
                        },
                        [
                            "https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/mm10_WGS_sample_R2.fastq.gz"
                        ]
                    ],
                    [
                        "testSample_42",
                        {
                            "sampleID": "testSample_42"
                        },
                        [
                            "https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/mm10_WGS_sample_R1.fastq.gz"
                        ]
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-12T14:18:28.101214532"
    },
    "Multiple Lanes Paired-End Data": {
        "content": [
            {
                "0": [
                    [
                        "testSample_42",
                        {
                            "sampleID": "testSample_42"
                        },
                        [
                            "testSample_42_R1.fastq.gz:md5,6c3b7e552d974c48660ac0b114200d25",
                            "testSample_42_R2.fastq.gz:md5,9ef301d9527a60d589b9b85072f52e68"
                        ]
                    ]
                ],
                "read_meta_ch": [
                    [
                        "testSample_42",
                        {
                            "sampleID": "testSample_42"
                        },
                        [
                            "testSample_42_R1.fastq.gz:md5,6c3b7e552d974c48660ac0b114200d25",
                            "testSample_42_R2.fastq.gz:md5,9ef301d9527a60d589b9b85072f52e68"
                        ]
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-12T14:18:43.874722493"
    },
    "Multiple Lanes Single-End Data": {
        "content": [
            {
                "0": [
                    [
                        "testSample_42",
                        {
                            "sampleID": "testSample_42"
                        },
                        [
                            "testSample_42_R1.fastq.gz:md5,a32fd77e38836b9b4ec0eb50818e52e8"
                        ]
                    ]
                ],
                "read_meta_ch": [
                    [
                        "testSample_42",
                        {
                            "sampleID": "testSample_42"
                        },
                        [
                            "testSample_42_R1.fastq.gz:md5,a32fd77e38836b9b4ec0eb50818e52e8"
                        ]
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-12T14:18:59.683040124"
    },
    "Paired-End Data": {
        "content": [
            {
                "0": [
                    [
                        "testSample_101",
                        {
                            "sampleID": "testSample_101"
                        },
                        [
                            "https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/human/hg38_WGS_sample_R1.fastq.gz",
                            "https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/human/hg38_WGS_sample_R2.fastq.gz"
                        ]
                    ],
                    [
                        "testSample_42",
                        {
                            "sampleID": "testSample_42"
                        },
                        [
                            "https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/mm10_WGS_sample_R1.fastq.gz",
                            "https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/mm10_WGS_sample_R2.fastq.gz"
                        ]
                    ]
                ],
                "read_meta_ch": [
                    [
                        "testSample_101",
                        {
                            "sampleID": "testSample_101"
                        },
                        [
                            "https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/human/hg38_WGS_sample_R1.fastq.gz",
                            "https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/human/hg38_WGS_sample_R2.fastq.gz"
                        ]
                    ],
                    [
                        "testSample_42",
                        {
                            "sampleID": "testSample_42"
                        },
                        [
                            "https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/mm10_WGS_sample_R1.fastq.gz",
                            "https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/mm10_WGS_sample_R2.fastq.gz"
                        ]
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-12T14:18:17.239226152"
    }
}